Admin/Test-AMSI.ps1

```diff
@@ -1,144 +1,793 @@
+                    Write-Host ""
+                    CheckAMSIConfig -ExchangeServer $server
+                    Write-Host ""
+                }
+            }
+            Write-Host $bar
+            Write-Host ""
+            $getSO = $null
+            $getSO = Get-SettingOverride -ErrorAction SilentlyContinue | Where-Object {
+                ( $_.SectionName -eq 'HttpRequestFiltering' ) -and
+                ( $_.Parameters -eq 'Enabled=False' ) -and
+                ( $null -eq $_.Server ) }
+            if ( $getSO ) {
+                $getSO | Out-Host
+                if ( $getSO.Status -eq "Accepted" ) {
+                    Write-Warning "AMSI is Disabled by $($getSO.Identity) SettingOverride at organization Level."
+                } else {
+                    Write-Host "We found SettingOverride for $ExchangeServer ($($getSO.Identity))"
+                    Write-Warning "The Status of $($getSO.Identity) is not Accepted. Should not apply at organization Level."
+                }
+            } else {
+                Write-Host "AMSI is Enabled for Exchange at Organization Level." -ForegroundColor Green
+                Write-Host "We did not find any Settings Override that remove AMSI at organization Level."
+                Write-Host ""
+            }
+        }
+
+        $needsRefresh = 0
+        if ($EnableAMSI) {
+            $getSO = $null
+            if ($filterList) {
+                foreach ( $server in $filterList ) {
+                    Write-Host $bar
+                    Write-Host ""
+                    $getSO = Get-SettingOverride -ErrorAction SilentlyContinue | Where-Object {
+                        ( $_.SectionName -eq 'HttpRequestFiltering' ) -and
+                        ( $_.Parameters -eq 'Enabled=False' ) -and
+                        ( $_.Server -contains $server ) }
+                    if ( $null -eq $getSO ) {
+                        Write-Host "We did not find Get-SettingOverride that disabled AMSI on $server"
+                        Write-Warning "AMSI is NOT disabled on $server"
+                    } else {
+                        Write-Host "Removing SettingOverride $($getSO.Identity)"
+                        $getSO | Out-Host
+                        Remove-SettingOverride -Identity $getSO.Identity -Confirm:$false
+                        Write-Host "Enabled on $server" -ForegroundColor Green
+                        Get-ExchangeDiagnosticInfo -Process Microsoft.Exchange.Directory.TopologyService -Component VariantConfiguration -Argument Refresh -Server $server | Out-Null
+                        Write-Host "Refreshed Get-ExchangeDiagnosticInfo on $server"
+                        $needsRefresh++
+                    }
+                }
+            } else {
+                Write-Host $bar
+                Write-Host ""
+                $getSO = Get-SettingOverride -ErrorAction SilentlyContinue | Where-Object {
```

Comment:
`Get-FilteredSettingOverrideInformation.ps1` could be useful here as well

Reply:
@iserrano76 do you want to check if `Get-FilteredSettingOverrideInformation` could be used here to simplify the code or is this something you plan to do in a later PR (or something that you not consider to use)?

Reply:
I did some test and I think Get-FilteredSettingOverrideInformation is a better solution but I was waiting to confirm if it is going to be included in shared and at the same time I need more time to fully understand and check how to include it.
Maybe another PR later is better.

Reply:
I'm fine with adding it via a new PR. Resolving this conversation for now.